ipa-core/src/protocol/step/mod.rs

```diff
@@ -36,10 +36,10 @@ pub trait StepNarrow<S: Step + ?Sized> {
 pub trait Step: AsRef<str> {}
 
 // In test code, allow a string (or string reference) to be used as a `Step`.
-#[cfg(any(feature = "test-fixture", debug_assertions))]
+#[cfg(feature = "descriptive-gate")]
```

Comment:
As I understand things (and possibly this can change with @martinthomson's new compact gate implementation), the compact gate supports only the steps that are part of the "production" IPA protocol. So anything that's not the production protocol needs descriptive gate. (Which is counter to my claim in my other comment that `bench` should work out-of-the-box for benchmarks -- but better to enable just descriptive gate, than descriptive gate plus every debug assertion.)  

I agree that we don't want a coverage gap in CI for steps added to the protocol without compact gate support -- but doesn't the "compact gate tests" (`cargo test --no-default-features --features "cli web-app real-world-infra test-fixture compact-gate"`) step in CI protect us from that? Not using a string as a step is necessary but not sufficient for a protocol to work with compact gate -- it also needs to be listed in steps.txt.

Reply:
is there a strong reason why you want to use strings instead of enums outside of tests? I just feel like it introduces more friction working on the code where you compile everything just fine and rely on CI to catch those issues which delays the feedback loop unnecessarily.

Reply:
No, no good reason. I had been conflating strings as steps with descriptive gate in my head. It's totally reasonable to implement `Step` for benchmarks.

Your comment prompted me to wonder whether there's a compelling reason we need to support strings as steps at all... I took a quick look at this and it seems like maybe not. The list of places we do (post IPAv2 removal) is relatively short, and a bunch of those places we're narrowing the context unnecessarily, so the narrow can just be removed. Elsewhere we just need to add an enum with `#[derive(Step)]`, which is a lot less work than when it required the enum definition plus a handful of impls for the enum.